src/gwt/src/org/rstudio/studio/client/common/mathjax/MathJax.java

```diff
@@ -300,12 +317,14 @@ private void renderLatexLineWidget(final Range range,
             @Override
             public void execute(LineWidget w)
             {
+               pendingLineWidgets_.remove(row);
```

Comment:
There are a number of failure modes here that could cause the row to get stuck in a permanently pending state such that we'd never render a line widget there again (e.g. if an exception is thrown anywhere in `createMathJaxLineWidget` or any of the other codepaths that execute to prepare for running the callback, or if the callback just doesn't get invoked). 

How about just iterating over the existing set of `PinnedLineWidget` objects and seeing if any are already bound to the given `row`?